### PR TITLE
chore: Update Python Otel dependencies

### PR DESCRIPTION
**Pull request by arbrown**
Consolidating a bunch of different OpenTelemetry updates into one because individually the renovate PRs failed (perhaps it didn't understand the order of the dependency chain, and was trying to do them separately and out of order.)

### CODE REVIEW

File: src/recommendationservice/requirements.in [hunk 1]
```diff
@@ -6,4 +6,4 @@ requests==2.31.0
 rsa==4.9
 opentelemetry-distro==0.39b0
 opentelemetry-instrumentation-grpc==0.39b0
```
Comment (bourgeoisor) [inline src/recommendationservice/requirements.in:8]: All `0.39b0` could be updated to `0.40b0`

File: src/emailservice/requirements.in [hunk 2]
```diff
@@ -8,4 +8,4 @@ google-cloud-trace==1.11.2
 requests==2.31.0
 opentelemetry-distro==0.39b0
 opentelemetry-instrumentation-grpc==0.39b0
```
Comment (bourgeoisor) [inline src/emailservice/requirements.in:10]: Same comment

**Review by bourgeoisor — CHANGES_REQUESTED** [review]

**Review by bourgeoisor — APPROVED** [review]
LGTM pending checks passing

**Review by NimJay — APPROVED** [review]

**Comment by NimJay** [comment]
Thank you, @arbrown, for doing this! (And thanks, @bourgeoisor, for the quick feedback/review!)
Merging.

**Pull request merged**